### PR TITLE
Light Theme

### LINKED ISSUE #1136: Add proper white theme for Cutter
Default Cutter configuration is to use the native theme (GTK, whatever, ...) and so our default color theme is not very pretty.
Although it can be switched in the panel options, I think that we should find a way to dark if the native theme is light (all Windows user native theme is light) or dark, and have a default theme for each one, because it should just be fine by default.

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Detailed description**
This PR implements a Light theme for Cutter. This was requested for awhile and finally it is here. I also refactored some behaviors to be more generic and implemented some methods for Cutter to be more in control and aware for its interface theme.

The light theme isn't perfect and still need to be tweaked and improve. Some controls are not satisfying me but I think that for now, it is good enough and ready for production, Changes will be added with time and by users feedback 

**Test plan (required)**

Interface:
image

Preferences:
image

Dashboard:
image

**Closing issues**

Closes #1136

### CODE REVIEW

**Comment by xarkes** [comment]
Damn it looks very good!

File: src/core/MainWindow.cpp [hunk 1]
```diff
@@ -1196,3 +1206,21 @@ void MainWindow::messageBoxWarning(QString title, QString message)
...
+{
+    // List of QActions which have alternative icons in different themes
+    const QList<QPair<void*, QString>> kSupportedIconsNames {
+        { ui->actionForward, QStringLiteral("arrow_right.svg") },
+        { ui->actionBackward, QStringLiteral("arrow_left.svg") },      
+    };
+
+
+    // Set the correct icon for the QAction
+	qhelpers::setThemeIcons(kSupportedIconsNames, [](void *obj, const QIcon &icon) {
```
Comment (xarkes) [inline src/core/MainWindow.cpp]: Indentation issues.

File: src/dialogs/preferences/PreferencesDialog.cpp [hunk 2]
```diff
@@ -99,3 +104,32 @@ void PreferencesDialog::changePage(QTreeWidgetItem *current, QTreeWidgetItem *pr
...
+        { QStringLiteral("Assembly"), QStringLiteral("disas.svg") },
+        { QStringLiteral("Graph"), QStringLiteral("graph.svg") },
+        { QStringLiteral("Appearance"), QStringLiteral("polar.svg") },
+        { QStringLiteral("Plugins"), QStringLiteral("plugins.svg") },
+        
+    };
+    QList<QPair<void*, QString>> supportedIconsNames;
+
+    foreach(const auto &p, kCategoryIconsNames)
+    {
```
Comment (xarkes) [inline src/dialogs/preferences/PreferencesDialog.cpp]: `{` on same line as foreach? Missing space after `foreach`

File: src/dialogs/preferences/PreferencesDialog.cpp [hunk 3]
```diff
@@ -99,3 +104,32 @@ void PreferencesDialog::changePage(QTreeWidgetItem *current, QTreeWidgetItem *pr
...
+        QList<QTreeWidgetItem *> items = ui->configCategories->findItems(p.first, Qt::MatchContains|Qt::MatchRecursive, 0);
+		if (items.isEmpty()) {
+			continue;
+        }
+        supportedIconsNames.append( { items.first(), p.second } );
+    }
+
+    // Set the correct icon for the QAction
+	qhelpers::setThemeIcons(supportedIconsNames, [](void *obj, const QIcon &icon) {
+		// here we have our setter functor, in this case it is almost "unique" because it specified the column in `setIcon` call
```
Comment (xarkes) [inline src/dialogs/preferences/PreferencesDialog.cpp]: Indentation issue

Comment (a1ext) [reply src/dialogs/preferences/PreferencesDialog.cpp]: Yeah, mixed spaces and tabs

**Pull request merged**